ui/components/src/test/render.tsx

```diff
@@ -0,0 +1,24 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { render, RenderOptions } from '@testing-library/react';
+import { QueryClient, QueryClientProvider } from 'react-query';
+
+const queryClient = new QueryClient({ defaultOptions: { queries: { refetchOnWindowFocus: false } } });
+
+/**
+ * Test helper to render a React component with some common app-level providers wrapped around it.
+ */
+export function renderWithContext(ui: React.ReactElement, options?: Omit<RenderOptions, 'queries'>) {
```

Comment:
it fiels weird to have this function in this package. Wouldn't it make more sense to have it in the `core` package ?

Reply:
Right, that makes sense to me since we also have the same function in `ui/dashboards/src/test/render.tsx`.

@sjcobb @LukeTillman  what do you think?

Reply:
Yeah, just to chime-in (sorry I've been a little incommunicado lately), I've been trying to keep `core` to just common data models/utils/etc. rather than anything React-specific, that way we can avoid having yet another package with a `react` library dependency. A test utils package might be another way to tackle sharing some of this type of stuff. Grafana does a similar thing with [one of their packages](https://www.npmjs.com/package/@grafana/e2e-selectors).

I didn't realize we'd need these common test things in multiple packages so quickly or I probably would've just set it up that way last week. I guess that's a good thing though since it means we're adding tests faster than I thought. 😂

Reply:
sounds good to me :).

Looking forward to see the improvement around this topic !